packages/app/src/components/inspector/PatternInspector.tsx

```diff
@@ -0,0 +1,238 @@
+  id,
+  images,
+  pattern,
+  onChangeImage,
+  onChangeFillType,
+  onChangeTileScale,
+  createImage,
+}: Props) {
+  const [showButton, setShowButton] = useState(false);
+  const [isDragOver, setIsDragOver] = useState(false);
+
+  const patternType = pattern.patternFillType;
+  const isTile = patternType === Sketch.PatternFillType.Tile;
+
+  const changeFillType = useCallback(
+    (value: PatternFillTypes) => {
+      onChangeFillType?.(Sketch.PatternFillType[value]);
+    },
+    [onChangeFillType],
+  );
+
+  const onSubmitTileScale = useCallback(
+    (value: number) => {
+      onChangeTileScale?.(value / 100);
+    },
+    [onChangeTileScale],
+  );
+
+  const onNudgeTileScale = useCallback(
+    (value: number) => {
+      onChangeTileScale?.(value / 100);
+    },
+    [onChangeTileScale],
+  );
+
+  const background = useMemo(
+    () => getPatternBackground(images, pattern.image),
+    [images, pattern.image],
+  );
+
+  const backgroundSize = useMemo(
+    () => getPatternSize(pattern.patternFillType, pattern.patternTileScale),
+    [pattern.patternFillType, pattern.patternTileScale],
+  );
+
+  const handleImageFile = useCallback(
+    async (file: File) => {
+      const data = await file.arrayBuffer();
+
+      const extension = file.type.split('/')[1];
+      const _ref = `images/${uuid()}.${extension}`;
+
+      createImage(data, _ref);
+      onChangeImage?.({
+        _class: 'MSJSONFileReference',
+        _ref: _ref,
+        _ref_class: 'MSImageData',
+      });
+    },
+    [createImage, onChangeImage],
+  );
+
+  const openFile = useCallback(async () => {
+    const file = await fileOpen({
+      extensions: imgFileExtensions.map((e) => `.${e}`),
+      mimeTypes: mimeTypes,
+    });
+
+    handleImageFile(file);
+  }, [handleImageFile]);
+
+  const handleDragEvent = useCallback((e: DragEvent, on?: boolean) => {
+    if (on) setIsDragOver(on);
+    e.preventDefault();
+    // Do something
+  }, []);
+
+  const handleDropEvent = useCallback(
+    async (e: DragEvent) => {
+      e.preventDefault();
+      for (const item of e.dataTransfer.items) {
+        // Careful: `kind` will be 'file' for both file
+        // _and_ directory entries.
+        if (item.kind === 'file' && mimeTypes.includes(item.type)) {
+          const entry = await item.getAsFile();
+
+          if (!entry) return;
+          handleImageFile(entry);
+        }
+      }
+      setIsDragOver(false);
+    },
+    [handleImageFile],
+  );
+
+  const scale = Math.round(pattern.patternTileScale * 100);
+  return (
+    <Column>
+      <Container
+        onDragOver={handleDragEvent}
+        onDragEnter={(e: DragEvent) => handleDragEvent(e, true)}
+        onDragLeave={(e: DragEvent) => handleDragEvent(e, false)}
+        onDrop={handleDropEvent}
+        isDragOver={isDragOver}
+        background={background}
+        backgroundSize={backgroundSize}
+        repeat={isTile}
+        onMouseEnter={() => setShowButton(true)}
```

Comment:
We have a `useHover` hook which we can use for this that handles some pointer weirdness. See:

https://github.com/noya-app/noya/blob/60d6525289203f0a325f805e7621ab50dc899261/packages/noya-designsystem/src/components/ListView.tsx#L178

We can pass a `onHoverChange` function to this hook to get updates about hover state